src/app/Authenticated.tsx

```diff
@@ -74,76 +77,86 @@ export const routeDetails = {
 };
 
 const Authenticated = () => {
-    return (
-        <Routes>
-            <Route
-                path={routeDetails.registration.path}
-                element={<Registration />}
-            />
-            <Route element={<AppLayout />}>
-                <Route path={routeDetails.home.path} element={<Home />} />
+    const { combinedGrants, isValidating } = useCombinedGrantsExt({
+        singleCall: true,
+    });
 
+    if (isValidating) {
+        return <FullPageSpinner />;
+    } else if (combinedGrants.length === 0) {
+        return <NoGrantsFound />;
+    } else {
```

Comment:
While checking for grants we want to show a spinner. Otherwise check if there are grants before showing the authenticated view.